plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/controller/Assets.js

```diff
@@ -0,0 +1,179 @@
+   */
+  showAssetInfo: function(container, componentModel, assetModel) {
+    var panel = container.down('nx-coreui-component-assetinfo'),
+        info = {};
+
+    if (!panel) {
+      panel = container.add({xtype: 'nx-coreui-component-assetinfo', weight: 10});
+    }
+
+    info[NX.I18n.get('BROWSE_ASSET_INFO_NAME')] = assetModel.get('name');
+    info[NX.I18n.get('BROWSE_ASSET_INFO_CONTENT_TYPE')] = assetModel.get('contentType');
+
+    panel.showInfo(info);
+  },
+
+  showComponentDetails: function(container, componentModel) {
+    var info1 = {}, info2 = {};
+
+    if (componentModel) {
+      info1[NX.I18n.get('BROWSE_SEARCH_ASSETS_REPOSITORY')] = componentModel.get('repositoryName');
+      info1[NX.I18n.get('BROWSE_SEARCH_ASSETS_FORMAT')] = componentModel.get('format');
+      info2[NX.I18n.get('BROWSE_SEARCH_ASSETS_GROUP')] = componentModel.get('group');
+      info2[NX.I18n.get('BROWSE_SEARCH_ASSETS_NAME')] = componentModel.get('name');
+      info2[NX.I18n.get('BROWSE_SEARCH_ASSETS_VERSION')] = componentModel.get('version');
+
+      container.down('#info1').showInfo(info1);
+      container.down('#info2').showInfo(info2);
+    }
+  },
+
+  /**
+   * @private
+   *
+   * Filter asset store based on component model.
+   *
+   * @param {NX.coreui.view.component.AssetList} grid assets grid
+   * @param {NX.coreui.model.Component} componentModel component owning the assets to be loaded
+   */
+  loadAssets: function(grid, componentModel) {
+    var assetStore = grid.getStore(),
+        filters;
+
+    if (componentModel) {
+      assetStore.clearFilter(true);
+      filters = [
+        {
+          property: 'repositoryName',
+          value: componentModel.get('repositoryName')
+        },
+        {
+          property: 'componentId',
+          value: componentModel.getId()
+        },
+        {
+          property: 'componentName',
+          value: componentModel.get('name')
+        }
+      ];
+      if (componentModel.get('group')) {
+        filters.push({
+          property: 'componentGroup',
+          value: componentModel.get('group')
+        });
+      }
+      if (componentModel.get('version')) {
+        filters.push({
+          property: 'componentVersion',
+          value: componentModel.get('version')
+        });
+      }
+      assetStore.addFilter(filters);
+    }
+  },
+
+  /**
+   * @private
+   *
+   * Update asset shown in asset container.
+   */
+  updateAssetContainer: function(gridView, td, cellIndex, assetModel) {
+    var me = this,
+        assetContainer = me.getAssetContainer();
+
+    assetContainer.refreshInfo(gridView.up('grid').componentModel, assetModel);
+  },
+
+  /**
+   * @private
+   *
+   * Show size in kb/mb/gb.
+   */
+  toSizeString: function(v) {
+    if (typeof v !== 'number') {
+      return NX.I18n.get('BROWSE_STANDARD_INFO_SIZE_UNKNOWN');
+    }
+    if (v < 0) {
+      return NX.I18n.get('BROWSE_STANDARD_INFO_SIZE_UNKNOWN');
+    }
+    if (v < 1024) {
```

Comment:
Code removed. It was copied over from NX2, not used by the moment. I left it in for when we will do the asset info, but your suggested Ext.util.Format.fileSize should do the job.